compound_disease/compound_treats_disease/word_vector_experiment/generate_word_vectors.py

```diff
@@ -0,0 +1,244 @@
+
+# coding: utf-8
+
+# # Generate Word Vectors For Compound Treats Disease Sentences
+
+# This notebook is designed to generate word vectors for gene interacts gene (CtD) sentences. Using facebooks's fasttext, we trained word vectors using all sentences that contain a disease and gene mention. The model was trained using the following specifications:
+# 
+# | Parameter | Value |
+# | --- | --- |
+# | Size | 300 |
+# | alpha | 0.005 | 
+# | window | 2 |
+# | epochs | 50 |
+# | seed | 100 | 
+
+# # Set Up Environment
+
+# In[2]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+from collections import defaultdict
+import os
+import pickle
+import sys
+
+sys.path.append(os.path.abspath('../../../modules'))
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+from tqdm import tqdm_notebook
+
+from gensim.models import FastText
+from gensim.models import KeyedVectors
+
+from utils.notebook_utils.dataframe_helper import load_candidate_dataframes, generate_embedded_df
+
+
+# In[3]:
+
+
+#Set up the environment
+username = "danich1"
+password = "snorkel"
+dbname = "pubmeddb"
+
+#Path subject to change for different os
+database_str = "postgresql+psycopg2://{}:{}@/{}?host=/var/run/postgresql".format(username, password, dbname)
+os.environ['SNORKELDB'] = database_str
+
+from snorkel import SnorkelSession
+session = SnorkelSession()
+
+
+# In[4]:
+
+
+from snorkel.learning.pytorch.rnn.rnn_base import mark_sentence
+from snorkel.learning.pytorch.rnn.utils import candidate_to_tokens
+from snorkel.models import Candidate, candidate_subclass
+
+
+# In[5]:
+
+
+CompoundDisease = candidate_subclass('CompoundDisease', ['Compound', 'Disease'])
+
+
+# # Compound Treats Disease
+
+# This section loads the dataframe that contains all compound treats disease candidate sentences and their respective dataset assignments.
+
+# In[9]:
+
+
+cutoff = 300
+total_candidates_df = (
+    pd
+    .read_table("../dataset_statistics/results/all_ctd_map.tsv.xz")
+    .query("sen_length < 300")
+)
+total_candidates_df.head(2)
```

Comment:
Is there a reason why the "disease" column is NaN.  This table is populated based on your tagger?

Reply:
Good catch. Turns out the file I generated in a previous notebook added an extra column called disease. This column is replaced with doid_name. I wrote down to remove this column in a future PR.